packages/@lwc/engine-dom/src/apis/build-custom-element-constructor.ts

```diff
@@ -45,20 +46,36 @@ export function deprecatedBuildCustomElementConstructor(
     return Ctor.CustomElementConstructor;
 }
 
+// Note: WeakSet is not supported in IE11, and the polyfill is not performant enough.
+//       This WeakSet usage is valid because this functionality is not meant to run in IE11.
+const hydratedCustomElements = new WeakSet<Element>();
+
 export function buildCustomElementConstructor(Ctor: ComponentConstructor): HTMLElementConstructor {
     const HtmlPrototype = getComponentHtmlPrototype(Ctor);
 
     return class extends HtmlPrototype {
         constructor() {
             super();
-            createVM(this, Ctor, {
-                mode: 'open',
-                owner: null,
-                tagName: this.tagName,
-            });
+
+            if (this.isConnected) {
+                // this if block is hit when there's already an un-upgraded element in the DOM with the same tag name.
+                hydrateComponent(this, Ctor, {});
```

Comment:
who else calls this method with an actual 3rd argument?

Reply:
`hydrateComponent` is part of the public api of lwc. any developer that wants to do hydration of an SSR rendered component. [rfc](https://github.com/salesforce/lwc-rfcs/blob/master/text/0117-ssr-rehydration.md)